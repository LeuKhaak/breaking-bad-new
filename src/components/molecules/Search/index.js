import React from "react";
import * as styles from "./styles.module.scss";
import Btn from "../../../components/atoms/Btn";
import loup from "src/assets/icons/loupe.svg";
import { Link } from "react-router-dom";

function Search({ createName, characterName, searchCharacter }) {
  return (
    <div className={styles.container}>
      <div className={styles.search}>
        <input
          className={styles.searchInput}
          type="text"
          name="search-card"
          placeholder="search"
          onChange={(event) => createName(event.target.value)}
          onKeyPress={(event) => searchCharacter(event)}
        />
        <Link to={`/person/${characterName}`} className={styles.link}>
          <Btn btnText="Find" icon={loup} />
        </Link>
      </div>
    </div>
  );
}

export default Search;
