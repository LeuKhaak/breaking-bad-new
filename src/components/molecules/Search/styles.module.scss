@import "../../../assets/styles/styles.scss";
.container {
  @include content;
  @include tabletSmall {
    width: 95%;
  }
  .search {
    display: flex;
    justify-content: space-between;
    align-items: center;
    max-width: 50rem;
    height: 3.5rem;
    background: $white;
    box-shadow: 0 0.25rem 0.5rem $shadowDark;
    border-radius: 0.25rem;
    margin: 0 auto 0 auto;
    @include mobileSmall {
      flex-direction: column;
      align-items: start;
      padding-top: 0.5rem;
    }
    .searchInput {
      border: none;
      outline: none;
      font-size: 1.25rem;
      padding-left: 1.25rem;
      &::placeholder {
        font-size: 1rem;
      }
    }
  }
}
