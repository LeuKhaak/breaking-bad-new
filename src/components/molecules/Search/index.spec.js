import React from "react";
import { shallow } from "enzyme";
import Search from ".";

let component = null;
const onChange = jest.fn();
const onKeyPress = jest.fn();
//const value = "test";
//const key = "Enter";
const setComponent = (props) => shallow(<Search {...props} />);
beforeEach(() => {
  component = setComponent();
});

describe("Search testing:", () => {
  test("is Search rendered", () => {
    component = setComponent({});
    expect(component).toMatchSnapshot();
  });

  test("onChangeFunc", () => {
    component = setComponent({
      createName: onChange,
      searchCharacter: onKeyPress,
    });
    component.find("input").simulate("change", { target: { value: "S" } });
    expect(onChange).toBeCalledWith(
      expect.objectContaining({ target: { value: "S" } })
    );
    //.expect("S");
    //component.find("input").simulate("change", { target: { value: "k" } });
    //expect(onChange).toBeCalledWith(expect("k"));
    //component.find("input").simulate("keydown", { keyCode: 13 });
    //expect(onKeyPress).toBeCalledWith(expect.objectContaining(expect("Sk")));
  });

  //test("onKeyPressFunc", () => {
  //  component = setComponent({
  //    onKeyPress: searchCharacter,
  //    icon: "./test.jpg",
  //  });
  //  component.find("input").simulate("keypress", { key, value });
  //  expect(searchCharacter).toBeCalledWith(
  //    expect.objectContaining({ key, value })
  //  );
  //});
});
