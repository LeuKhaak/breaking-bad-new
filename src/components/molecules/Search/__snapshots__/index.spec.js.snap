// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search testing: is Search rendered 1`] = `
<div>
  <div>
    <input
      name="search-card"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="search"
      type="text"
    />
    <Link
      to="/person/undefined"
    >
      <Btn
        btnText="Find"
        icon="test-file-stub"
      />
    </Link>
  </div>
</div>
`;
